### PR TITLE
[6.2.z] cherry-pick test rex connect by ip

### PR DESCRIPTION
**Pull request by pondrejk**
Cherry-pick from #4648
Related issue: #4624 
Test results:

```nosetests -v tests/foreman/cli/test_remoteexecution.py:RemoteExecutionTestCase.test_positive_run_custom_job_template_by_ip
Run custom template on host connected via ip ... ok

----------------------------------------------------------------------
Ran 1 test in 304.554s

OK

 nosetests -v tests/foreman/cli/test_remoteexecution.py:RemoteExecutionTestCase.test_positive_run_default_job_template_by_ip
Run default template on host connected via ip ... ok

----------------------------------------------------------------------
Ran 1 test in 299.689s

OK
```

### CODE REVIEW

File: tests/foreman/cli/test_remoteexecution.py [hunk 1]
```diff
@@ -316,6 +317,75 @@ def test_positive_run_custom_job_template(self):
...
+
+        @id: 811c7747-bec6-4a2d-8e5c-b5045d3fbc0d
+
+        @expectedresults: Verify the job was successfully ran against the host
+        """
+        Settings.set({
+            'name': 'remote_execution_connect_by_ip',
+            'value': 'True',
+        })
+        invocation_command = make_job_invocation({
```
Comment (oshtaier) [inline tests/foreman/cli/test_remoteexecution.py]: Maybe we should put next statements into `try...finally` block to return settings value no matter what result we get?

Comment (pondrejk) [inline tests/foreman/cli/test_remoteexecution.py]: @oshtaier addressed, ok with this form?

**Comment by pondrejk** [comment]
Hello, after some research and discussion with @rplevka, I decided to take a different approach to these tests (using per-host setting instead of global setting). Therefore I put do_not_merge. Also this can help us to get more rex tests running in automation until the problems from #4578 are resolved

**Comment by pondrejk** [comment]
@renzon @oshtaier ready for review again, I enable the remote_execution_connect_by_ip per host, I added more tests -- these are by_ip version of existing tests. Tests:

```
py.test tests/foreman/cli/test_remoteexecution.py -k _by_ip 
...
=================== 5 passed, 1 skipped, 15 deselected in 1128.98 seconds ====================
```

**Comment by renzon** [comment]
ACK

**Pull request merged**